compiler/resolution/functionResolution.cpp

```diff
@@ -2506,6 +2502,61 @@ static FnSymbol* wrapAndCleanUpActuals(ResolutionCandidate* best,
                                        CallInfo&            info,
                                        bool                 followerChecks);
 
+static bool isTypeConstructionCall(CallExpr* call) {
+  bool ret = false;
+
+  if (SymExpr* se = toSymExpr(call->baseExpr)) {
+    if (se->symbol()->hasFlag(FLAG_TYPE_VARIABLE)) {
+      if (se->typeInfo()->symbol->hasFlag(FLAG_TUPLE) &&
+          se->typeInfo() != dtTuple) {
```

Comment:
Hmm.. at first I thought that this exception had to do with type methods for tuples but now I'm not so sure. Because I think that the baseExpr normally refers to a FnSymbol or an UnresolvedSymExpr. What are we doing with tuples where the baseExpr is an instantiated tuple type?

At the least I think it deserves a TODO comment as it looks like a workaround to me.

Reply:
In this PR a baseExpr can also now be a SymExpr to a TypeSymbol. I believe this can happen with instantiated tuples-of-types when accessing its components.

Reply:
Right, but why wouldn't the instantiated tuple of types end up having a baseExpr that is the `this` method, e.g.?